lib/Target/TfheRustBool/TfheRustBoolEmitter.cpp

```diff
@@ -215,9 +245,25 @@ void TfheRustBoolEmitter::emitReferenceConversion(Value value) {
 LogicalResult TfheRustBoolEmitter::printSksMethod(
     ::mlir::Value result, ::mlir::Value sks, ::mlir::ValueRange nonSksOperands,
     std::string_view op, SmallVector<std::string> operandTypes) {
-  if (isa<TensorType>(nonSksOperands[0].getType())) {
+  // ToDo: Check again the enum definition
```

Comment:
You should be able to check whether the op str can be symbolized as an enum
```
if (packedAPI && symbolizeTfheRustBoolGateEnum(op).has_value())
```

At least, each enum attr should generate `symbolizeENUMNAME` function that returns a `std::optional`

Reply:
Thanks! Was looking for something like this!